meteor/client/ui/RundownList/RundownShelfLayoutSelection.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react'
+import { Translated, translateWithTracker } from '../../lib/ReactMeteorData/ReactMeteorData'
+import { UIStateStorage } from '../../lib/UIStateStorage'
+import { Link } from 'react-router-dom'
+import { SplitDropdown } from '../../lib/SplitDropdown'
+import { getRundownPlaylistLink, getRundownWithLayoutLink, getShelfLink } from './util'
+import { Rundown } from '../../../lib/collections/Rundowns'
+import { RundownLayoutBase, RundownLayouts } from '../../../lib/collections/RundownLayouts'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import classNames from 'classnames'
+import { IconProp } from '@fortawesome/fontawesome-svg-core'
+
+interface IRundownShelfLayoutSelectionProps {
+	rundown: Rundown
+}
+
+interface IRundownShelfLayoutSelectionTrackedProps {
+	rundownLayouts: RundownLayoutBase[]
+}
+
+interface IRundownShelfLayoutSelectionState {
+	selectedView: string
+}
+
+export const RundownShelfLayoutSelection = translateWithTracker<
+	IRundownShelfLayoutSelectionProps,
+	IRundownShelfLayoutSelectionState,
+	IRundownShelfLayoutSelectionTrackedProps
+>((props: IRundownShelfLayoutSelectionProps) => {
+	const rundownLayouts = RundownLayouts.find({ showStyleBaseId: props.rundown.showStyleBaseId }).fetch()
+
+	return {
+		...props,
+		rundownLayouts,
```

Comment:
1. RundownLayouts are already passed as a prop into RundownPlaylistUi (RundownList.tsx:254). I don't see a point in registering a tracker here and a tracker for the RundownLayouts there. Considering the general structure of all of the RundownList compoentents, I would suggest removing this tracker and relying on the list of RundownLayouts passed down through props from the main tracker.
2. You don't have to (and generally shouldn't) return all of the incoming props back into a `translateWithTracker`. The HOC will merge whatever you return from this tracker function with incoming props and inject that into the child component.